lang/cs/Org.Apache.REEF.Network/Elastic/Driver/Default/DefaultElasticStage.cs

```diff
@@ -0,0 +1,509 @@
+        /// </summary
+        public IElasticContext Context { get; private set; }
+
+        /// <summary>
+        /// Whether the stages contains iterations or not.
+        /// </summary>
+        public bool IsIterative { get; set; }
+
+        /// <summary>
+        /// The failure state of the target stages.
+        /// </summary>
+        public IFailureState FailureState { get; private set; }
+
+        /// <summary>
+        /// Whether the stages is completed or not.
+        /// </summary>
+        public bool IsCompleted
+        {
+            get { return FailureState.FailureState == (int)DefaultFailureStates.Complete; }
+        }
+
+        /// <summary>
+        /// Generates an id to uniquely identify operators in the stages.
+        /// </summary>
+        /// <returns>A new unique id</returns>
+        public int GetNextOperatorId()
+        {
+            return Interlocked.Increment(ref _numOperators);
+        }
+
+        /// <summary>
+        /// Add a partitioned dataset to the stage.
+        /// </summary>
+        /// <param name="inputDataSet">The partitioned dataset</param>
+        /// <param name="isMasterGettingInputData">Whether the master node should get a partition</param>
+        public void AddDataset(IPartitionedInputDataSet inputDataSet, bool isMasterGettingInputData = false)
+        {
+            AddDataset(inputDataSet.Select(x => x.GetPartitionConfiguration()).ToArray(), isMasterGettingInputData);
+        }
+
+        /// <summary>
+        /// Add a set of datasets to the stage.
+        /// </summary>
+        /// <param name="inputDataSet">The configuration for the datasets</param>
+        /// <param name="isMasterGettingInputData">Whether the master node should get a partition</param>
+        public void AddDataset(IConfiguration[] inputDataSet, bool isMasterGettingInputData = false)
+        {
+            _isMasterGettingInputData = isMasterGettingInputData;
+
+            _datasetConfiguration = Optional<IConfiguration[]>.Of(inputDataSet);
+        }
+
+        /// <summary>
+        /// Finalizes the stages.
+        /// After the stages has been finalized, no more operators can
+        /// be added to the group.
+        /// </summary>
+        /// <returns>The same finalized stages</returns>
+        public IElasticStage Build()
+        {
+            if (_finalized == true)
+            {
+                throw new IllegalStateException("Stage cannot be built more than once");
+            }
+
+            if (_datasetConfiguration.IsPresent())
+            {
+                var adjust = _isMasterGettingInputData ? 0 : 1;
+
+                if (_datasetConfiguration.Value.Length + adjust < _numTasks)
+                {
+                    throw new IllegalStateException(
+                        $"Dataset is smaller than the number of tasks: "
+                        + "re-submit with {_datasetConfiguration.Value.Length + adjust} tasks");
```

Comment:
put `$` before the second string, otherwise it won't get interpolated

Reply:
Ops. Fixed.